packages/@aws-cdk/core/lib/stack.ts

```diff
@@ -704,15 +717,27 @@ export class Stack extends Construct implements ITaggable {
       return;
     }
 
-    const deps = this.dependencies.map(s => s.stackName);
+    const deps = this.dependencies.map(s => s.artifactId);
     const meta = this.collectMetadata();
 
+    // backwards compatibility since originally artifact ID was always equal to
+    // stack name the stackName attribute is optional and if it is not specified
+    // the CLI will use the artifact ID as the stack name. we *could have*
+    // always put the stack name here but wanted to minimize the risk around
+    // changes to the assembly manifest. so this means that as long as stack
+    // name and artifact ID are the same, the cloud assembly manifest will not
+    // change.
+    const stackNameProperty = this.stackName === this.artifactId
```

Comment:
This seems overly defensive. Adding new fields with new values should always be safe.

Reply:
yes, see comment above - trying to avoid test expectation update churn